core/domain/stats_jobs_one_off.py

```diff
@@ -1581,3 +1581,72 @@ def reduce(key, values):
             yield (key, len(values))
         else:
             yield (key, values)
+
+
+class ResetExplorationIssuesOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """A one-off job to reset all ExplorationIssues to a valid/empty state."""
+
+    REDUCE_KEY_DELETED = 'Referenced PlaythroughModel(s) deleted'
+    REDUCE_KEY_MISSING = 'Missing ExplorationIssuesModel(s) regenerated'
+    REDUCE_KEY_RESET = 'Existing ExplorationIssuesModel(s) reset'
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @staticmethod
+    def map(latest_exp_model):
+        exp_id = latest_exp_model.id
+        exp_versions = list(python_utils.RANGE(1, latest_exp_model.version + 1))
+
+        issues_model_cls = stats_models.ExplorationIssuesModel
+        issues_models = issues_model_cls.get_multi(
+            [issues_model_cls.get_entity_id(exp_id, v) for v in exp_versions],
+            include_deleted=True)
+
+        num_models_reset = 0
+        playthrough_ids_referenced = set()
+        exp_versions_without_models = list()
+
+        for i, issues_model in enumerate(issues_models):
+            if issues_model is not None:
+                for issue in issues_model.unresolved_issues:
+                    playthrough_ids_referenced.update(issue['playthrough_ids'])
+                issues_model.unresolved_issues = []
+                num_models_reset += 1
+            else:
+                issues_models[i] = issues_model_cls(
+                    id=issues_model_cls.get_entity_id(exp_id, exp_versions[i]),
+                    exp_id=exp_id, exp_version=exp_versions[i],
+                    unresolved_issues=[])
+                exp_versions_without_models.append(exp_versions[i])
+
+        issues_model_cls.update_timestamps_multi(issues_models)
+        issues_model_cls.put_multi(issues_models)
+
+        if playthrough_ids_referenced:
+            stats_models.PlaythroughModel.delete_multi(
+                stats_models.PlaythroughModel.get_multi(
+                    playthrough_ids_referenced))
```

Comment:
Will this fail if the playthrough IDs aren't valid/are dangling references? Suggest adding a test for that case (if it's not already covered in the existing tests).

Reply:
Done.